contracts/margin/external/BucketLender/BucketLender.sol

```diff
@@ -1193,16 +1193,21 @@ contract BucketLender is
         view
         returns (uint256)
     {
-        assert(!Margin(DYDX_MARGIN).isPositionClosed(POSITION_ID));
+        // load variables from storage;
+        Margin margin = Margin(DYDX_MARGIN);
+        bytes32 positionId = POSITION_ID;
+        uint32 bucketTime = BUCKET_TIME;
+
+        assert(!margin.isPositionClosed(positionId));
 
         // if position not created, allow deposits in the first bucket
-        if (!Margin(DYDX_MARGIN).containsPosition(POSITION_ID)) {
+        if (bucketTime == 0 || !margin.containsPosition(positionId)) {
```

Comment:
I don't think I agree with this fix. Why not just disallow bucket lenders to be deployed with `bucketTime == 0` in the first place?

This could be a common error as web3 will just pass in 0 if you forget to pass in the argument. I've already done this myself.

It's also very unclear what even the expected behavior is if `bucketTime` is 0: Are there infinite buckets? Is there just one bucket? Does the bucket time just go up to 1 second?

Finally, it seems like having a `bucketTime` of 0 could lead to edge case bugs such as this.